src/Visualization/Python/InterpolateVolumeData.py

```diff
@@ -0,0 +1,342 @@
+                interpolated_tensor = interpolant.interpolate(component_data)
+                tensor_path = "{}/{}".format(grid_name, tensor_names[j])
+                tensor_comps.append(
+                    TensorComponent(
+                        tensor_path, DataVector(interpolated_tensor,
+                                                copy=False)))
+
+            volume_data.append(
+                ElementVolumeData(target_mesh.extents(), tensor_comps,
+                                  target_mesh.basis(),
+                                  target_mesh.quadrature()))
+        target_vol = target_file.get_vol(target_volume_data)
+        target_vol.write_volume_data(obs, obs_value, volume_data)
+
+    source_file.close()
+    if not target_file is source_file:
+        target_file.close()
+
+
+def parse_args(sys_args):
+    """
+    defines and parses the command line arguments
+
+    Parameters
+    ----------
+    sys_args arguments passed via command line
+
+    Returns
+    -------
+    A dictionary of the parsed command line arguments
+    """
+    parser = argparse.ArgumentParser(
+        description=
+        "Interpolate the tensor components of h5 files to a desired grid")
+
+    parser.add_argument(
+        "--source-file-prefix",
+        required=True,
+        help=(
+            "The prefix for the .h5 source files. All files starting with the "
+            "prefix followed by a number will be interpolated."))
+
+    parser.add_argument("--source-subfile-name",
+                        required=True,
+                        help=("The name of the volume data subfile within the "
+                              "source files in which the data is contained"))
+
+    parser.add_argument(
+        "--target-file-prefix",
+        default=None,
+        help=
+        ("The prefix for the target files where the interpolated data is "
+         "written. When no target file is specified, the interpolated data is "
+         "written to the corresponding source file in a new volume data "
+         "subfile."))
+
+    parser.add_argument(
+        "--target-subfile-name",
+        required=True,
+        help=("The name of the volume data subfile within the target "
+              "files where the data will be written."))
+
+    parser.add_argument(
+        "--tensor-components",
+        nargs='*',
+        help=("The names of the tensors that are to be interpolated. "
+              "Accepts regular expression. "
+              "If none are specified, all tensors are interpolated."))
+
+    parser.add_argument(
+        "--target-extents",
+        type=int,
+        nargs='+',
+        required=True,
+        help=("The extents of the target grid. "
+              "Can be different for each dimension e.g. '3 5 4'"))
+
+    parser.add_argument("--target-basis",
+                        nargs='+',
+                        choices=basis_dict.keys(),
+                        default=["Legendre"],
```

Comment:
Better remove this default, and the one for `--target-quadrature` as well (I would expect the default to keep the basis/quadrature unchanged from the source file, not to always interpolate to LGL grids).